provisioning/miniprov/hare_mp/main.py

```diff
@@ -310,39 +310,51 @@ def kv_cleanup():
         raise RuntimeError('Error during key delete in transaction')
 
 
+def pre_factory():
+    logging.info('Executing pre-factory cleanup command...')
+    deployment_logs_cleanup()
+
+
 def cleanup(args):
     try:
-        rc = -1
+        kv_cleanup()
+        logs_cleanup()
+        config_cleanup()
+        if args.pre_factory:
+            pre_factory()
 
-        if kv_cleanup() == 0 and config_cleanup() == 0 and logs_cleanup() == 0:
-            rc = 0
-
-        exit(rc)
+        exit(0)
     except Exception as error:
         logging.error('Error during cleanup (%s)', error)
         exit(-1)
 
 
 def logs_cleanup():
     try:
-        logging.info('Cleaning up hare log directory(/var/log/seagate/hare)')
-        os.system('rm -rf /var/log/seagate/hare')
+        logging.info('Cleaning up hare log directory(/var/log/seagate/hare/*)')
+        os.system('rm -f /var/log/seagate/hare/*')
 
-        return 0
     except Exception as error:
-        logging.error('Error during logs cleanup (%s)', error)
-        return -1
+        raise RuntimeError(f'Error during logs cleanup : key={error}')
 
 
 def config_cleanup():
     try:
         logging.info('Cleaning up hare config directory(/var/lib/hare)')
         os.system('rm -rf /var/lib/hare/*')
 
-        return 0
     except Exception as error:
-        logging.error('Error during config cleanup (%s)', error)
-        return -1
+        raise RuntimeError(f'Error during config cleanup : key={error}')
+
+
+def deployment_logs_cleanup():
+    try:
+        logging.info('''Cleaning up hare deployment log directory
+            (/var/log/seagate/hare/hare_deployment)''')
+        os.system('rm -rf /var/log/seagate/hare/hare_deployment')
```

Comment:
Another improvement that I see is using `os.remove(), os.unlink() or shutil.rmtree()` instead of a direct call to unix shell command. I think using python functions instead would increase the portability factor of the code.
What is your opinion @knekrasov ? cc @vaibhavparatwar